src/ajax/check_hgvs_dialogue.php

```diff
@@ -0,0 +1,603 @@
+        }
+
+        if (isset($aMappedVariant['warnings']['WFLAG'])) {
+            // This type of warning tells us that VariantValidator had a problem
+            //  which is an issue with them, not us nor our user. We can only get
+            //  the mapping on all genome builds, not on (other) transcripts.
+            // We will notify the user.
+            update_dialogue('<br>Your variant could not fully be validated due to unknown issues.');
+            // Fixme; Either find a fix within VV, or Call Mutalyzer.
+        }
+
+    // Check whether the mapping was successful.
+    } elseif (!isset($aMappedVariant['data']['DNA'])
+        || empty($aMappedVariant['data']['DNA'])) {
+        // Although we did not receive any warnings or errors, the DNA field
+        //  is left empty. This means we have no information on the mapping
+        //  and there is not much we can do... We will inform the user that
+        //  an unknown error occurred and that they should try again later.
+        update_images_per_step($sStepMapping, $sImageFailed);
+        update_dialogue(
+            '<br>An unknown error occurred while trying to validate and map your variant.' .
+            ' We are sorry for the inconvenience. Please try again later.',
+            $sButtonOKInvalid
+        );
+        exit;
+    }
+
+
+    // When sending in a variant on transcript, VariantValidator only
+    //  returns the variant as mapped on that one transcript. If we are
+    //  on a VOT creation form, and there are multiple transcripts open,
+    //  we want each transcript to get a mapping. To get this, we then
+    //  need to call VariantValidator a second time using one of the
+    //  genomic variant as were returned using our first call.
+    if ($sType == 'VOT' && count($aTranscripts) > 1) {
+
+        $aMappedViaGB = (
+        !isset($aMappedVariant['data']['genomic_mappings']['hg38']) ? // Yes=We have a genomic reference from our first call; No=We don't have a genomic reference.
+            array() :
+            $_VV->verifyGenomic($aMappedVariant['data']['genomic_mappings']['hg38'], array(
+                'map_to_transcripts' => true,          // Should we map the variant to transcripts?
+                'predict_protein'    => true,          // Should we get protein predictions?
+                'lift_over'          => false,         // Should we get other genomic mappings of this variant?
+                'select_transcripts' => array_diff($aTranscripts, array($sReferenceSequence)), // Should we limit our output to only a certain set of transcripts?
+            ))
+        );
+
+        if (!isset($aMappedViaGB['data']['transcript_mappings'])) {
+            // If for any reason no genomic mappings were given, we cannot perform this
+            //  extra step, and will thus miss some information. We will inform the user.
+            update_dialogue('<br>Your variant could not be mapped to all transcripts due to unknown issues.');
```

Comment:
An issue might lie here:
If a variant IS mapped to some extend, but there are fields which are left empty, the user has no way of filling these fields (since this script resets all mapped values when it is called again).

Reply:
So, basically, we should fill something in those fields (like `c.?`), and perhaps disable the `onchange()` because it won't do any good?

Reply:
After thinking about this some more, I guess removing the `onchange()` should only be done maybe if we're sure the transcript is the problem? Do we have examples of this situation?

Reply:
Filling in empty fields seems right. Perhaps I could add the option to run the dialogue on one variant only? If this options would be turned on, the script would no longer reset all values, and would also not perform mapping (VariantValidator would only be run to validate the variant's own syntax). Adding this option might also prove useful when LOVD will allow non-human genome builds.